sct-commons/src/main/java/org/lfenergy/compas/sct/commons/SclService.java

```diff
@@ -196,13 +219,70 @@ public List<SclReportItem> updateDoInRef(SCL scd) {
                 .toList();
     }
 
+
     @Override
-    public List<SclReportItem> manageMonitoringLns(SCL scd) {
-        SclRootAdapter sclRootAdapter = new SclRootAdapter(scd);
-        return sclRootAdapter.streamIEDAdapters()
-                .filter(iedAdapter -> !iedAdapter.getName().contains(IED_TEST_NAME))
-                .map(IEDAdapter::manageMonitoringLns)
-                .flatMap(List::stream)
+    public List<SclReportItem> manageMonitoringLns(SCL scd, JAXBContext jaxbContext) {
+        errorHanlder.clear();
+        iedService.getFilteredIeds(scd, ied -> !ied.getName().contains(IED_TEST_NAME))
+                        .forEach(tied -> {
+                            List<TExtRef> tExtRefs = ldeviceService.getLdevices(tied)
+                                    .flatMap(tlDevice -> extRefReaderService.getExtRefs(tlDevice.getLN0()))
+                                    .filter(tExtRef -> tExtRef.isSetServiceType() && tExtRef.isSetSrcCBName()
+                                            && (tExtRef.getServiceType().equals(TServiceType.GOOSE) || tExtRef.getServiceType().equals(TServiceType.SMV)))
+                                    .toList();
+                            if(!tExtRefs.isEmpty()){
+                               ldeviceService.findLdevice(tied, LDEVICE_LDSUIED)
+                                       .ifPresentOrElse(tlDevice -> {
+                                           manageMonitoringLns(tExtRefs.stream().filter(tExtRef -> TServiceType.GOOSE.equals(tExtRef.getServiceType())).toList(), scd, tied, tlDevice, DO_GOCBREF, MonitoringLnClassEnum.LGOS, jaxbContext);
+                                           manageMonitoringLns(tExtRefs.stream().filter(tExtRef -> TServiceType.SMV.equals(tExtRef.getServiceType())).toList(), scd, tied, tlDevice, DO_SVCBREF, MonitoringLnClassEnum.LSVS, jaxbContext);
+                                       }, ()-> errorHanlder.add(SclReportItem.warning(tied.getName(), String.format(MESSAGE_LDEVICE_INST_NOT_FOUND, LDEVICE_LDSUIED, tied.getName()))));
```

Comment:
no; it's a blocking problem, please use `orElseThrow`

Reply:
In the old version, it's not.
![image](https://github.com/user-attachments/assets/621d514f-9e9a-4960-a07c-0eb978a5111d)